packages/core/src/interpreter.ts

```diff
@@ -1349,13 +1348,24 @@ export function interpret<
 >(
   machine: StateMachine<TContext, TStateSchema, TEvent, TTypestate>,
   options?: Partial<InterpreterOptions>
-) {
-  const interpreter = new Interpreter<
+): Interpreter<TContext, TStateSchema, TEvent, TTypestate> {
+  const interpreter = new PrivateInterpreter<
     TContext,
     TStateSchema,
     TEvent,
     TTypestate
   >(machine, options);
 
-  return interpreter;
+  return interpreter as any;
+}
+
+export declare class Interpreter<
```

Comment:
This removes `Interpreter` class from being available as runtime value - we have never encouraged using it directly but it could be a breaking change for somebody.

I couldn't figure out how to cast a class properly while preserving both its type and the runtime value (since classes in TS are both at the same time). It almost feels like it would be easier to just add `Symbol.observable` to the `Interpreter` once `.d.ts` gets emitted - it seems hacky as f but would fix a lot of annoyances here.